core/foundation/res/ROOT/FoundationUtils.hxx

```diff
@@ -0,0 +1,37 @@
+/// \file FoundationUtils.hxx
+///
+/// \brief The file contains utilities which are foundational and could be
+/// across the core component of ROOT.
+///
+///
+/// \author Vassil Vassilev <vvasilev@cern.ch>
+///
+/// \date June, 2019
+///
+/*************************************************************************
+ * Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_CORE_FOUNDATION_FOUNDATIONUTILS_HXX
+#define ROOT_CORE_FOUNDATION_FOUNDATIONUTILS_HXX
+
+#include <string>
+
+namespace ROOT {
+namespace FoundationUtils {
+
+   ///\returns the $PWD.
+   void GetCurrentDir(std::string &output);
+
+   ///\returns the relative path of \c path with respect to \c base.
+   std::string MakePathRelative(const std::string &path, const std::string &base,
```

Comment:
Can you give an example? I think that would make the interface a lot clearer. Is this:
```
E.g. for path being "/a/b/c/d" and base "/a/b", returns "c/d"
```